### ISSUE TITLE
Recurring volume snapshot strange behaviour

### ISSUE THREAD

**Issue by cdfgallo** [apache/cloudstack#9128]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.19.0.1
~~~

##### OS / ENVIRONMENT

Master: CentOS7
Cluster: vSphere 7.0U3

##### SUMMARY

Recurring weekly snapshots taken from data disks on one VMware instance are not shown in the UI.
We set the following snapshot policy for a data volume of an instance and we observe some strange behaviours
![image](https://github.com/apache/cloudstack/assets/110984295/d39fd557-d4c4-41f5-b46d-9fc80a07db52)
We also have a daily recurring snapshot on the root disk (about 30GB) of the VM which starts at 10:30 PM (GMT +1).
While the snapshot on the root disks take an incredible amount of time to end (many times more than 2 hours), the snapshot of the data disk failed with error: "There is other active snapshot tasks on the instance to which the volume is attached, please try again later".
After that we can see that the snapshot on the data disk is taken and is correctly present on the secondary storage, but we can't see it in the UI.

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~
If a snapshot fails on the prerequisites it should not be performed and if it's performed it should be present in the UI.
~~~

##### ACTUAL RESULTS

~~~
From the UI we see that the snapshot fails, but it's performed and not showed in the UI
~~~

**Comment by rajujith** [comment]
@cdfgallo  could you update the time for the data volume snapshot schedule to 2 to 3 hours later than the ROOT volume?

**Comment by cdfgallo** [comment]
Hi @rajujith yes, separating the two workloads we are albe to process this.
I see anyway that the template export is quite slow in cloudstack and analyzing a little bit I see that, after creating a full clone of the VM, a OVA it's exported to the NFS mountpoint on the master node. It's the correct behaviour or I have some wrong configuration?

**Comment by rajujith** [comment]
@cdfgallo Yes, you understand the process correctly. CloudStack implements full backup for Vmware volume snapshot backup to the secondary storage.

**Comment by cdfgallo** [comment]
> @cdfgallo Yes, you understand the process correctly. CloudStack implements full backup for Vmware volume snapshot backup to the secondary storage.

@rajujith  Thank you! Wouldn't be easier to export it directly to the datastore mounted on the vCenter server instead of using the CS master node? The process would be at least, way faster than the actual.
